-chests-all-monsters,  -loot,  and -no-random-encounters

HansGR opened the pull request:
Three new flags:
--chests-all-monsters,  usage -cam <boss%>:  replaces all chests with Monsters-in-a-box, with <boss%> = [0--100] percentage of being a boss.  Non-bosses include normal MiaBs and randomly selected formations.
--loot,  usage -loot:  randomizes drops and steals for all enemies.  No control flags currently, 100% random from all items.
--no-random-encounters,  usage -nre:  sets encounter rate to 0 in all maps.  Changes Moogle Charm to Moogle Curse: when equipped, allows encounters on world map only at a low rate.

HansGR committed:
Loot (-loot) and Chests-All-MIAB (-cam)

First commit, added flags for randomize steals and drops (-loot) and make all chests monster-in-a-boxes (-cam).

Note that there's currently no selection on what can be in boxes, you will get bosses.

HansGR committed:
No Random Encounters (-nre)

self-explanatory.  All encounter rates = 0.  For use with Chests All MiaB and Loot.

HansGR committed:
bufgix - now no random encounters in dungeons

HansGR committed:
Add Moogle Curse

The Moogle Curse (reskin of Moogle Charm) draws random encounters on the world map, when equipped. It does not work in dungeons.  This is a workaround so you can still complete Gau and Gogo checks.  The encounter rate is very low to avoid abuse.

HansGR committed:
Exclude buggy encounters

Removes empty and bugged event battle groups from the MIAB encounter pool.  (Doesn't currently overwrite them.)

HansGR committed:
Make each MiaB unique; take over unused event_battle_groups

- Changed pad_enemy_packs to only put a single random formation in each one.  This would make races fairer.
- Expanded the event_battle_groups rewrite to claim those unused in FF6WC.

This leaves 54 bosses, 199 single normal enemy formations (for MiaB), and 3 untouched [Zone Eater, Tier 1, Final Kefka].

HansGR committed:
Add boss% to -chests-all-monsters

Now called as -cam <0--100>.  The number is the percentage chance for each box to contain a boss (0 = no bosses, 100 = all bosses).

asilverthorn commented on data/data.py:
```diff
@@ -38,8 +38,11 @@ def __init__(self, rom, args):
         self.enemies.mod(self.maps)
+        #for p in self.enemies.packs.packs:
+        #    forms = [(f, self.enemies.formations.get_name(f)) for f in p.formations]
+        #    print(p.id, ': ', forms)
```
minor nit: please remove commented out code

asilverthorn commented on data/items.py:
```diff
@@ -187,6 +188,20 @@ def moogle_starting_equipment(self):
+        # new_descr = "No random enemy encounters<end>"
+        new_descr = "Draw monsters on world map<end>"
+        name_bytes = text.get_bytes(new_descr, text.TEXT2)
+        self.rom.set_bytes(0x2d6400 + offset, name_bytes)
```
Please structure the setting of an item description like this (make a desc_data array in items.py and set the description in item.py):

https://github.com/ff6wc/WorldsCollide/pull/28/files#diff-5991f3562b7b58a2c5931c6a2897859699dea4634f1e33cd7511655c86c1165d

asilverthorn commented on event_battle_groups.txt:
```diff
@@ -0,0 +1,83 @@
+Formations to avoid:
```
is this file needed? Looks like it's the same data as enemy_battle_groups.py

HansGR replied:
This file is not needed. I don't know how to remove it from the branch though. Do I just delete the file and then push a new commit?

HansGR replied:
OK, I removed it.

HansGR committed:
address asilverthorn's comments

for PR to WC

HansGR commented:
All comments addressed.  -DT

HansGR committed:
Change -loot to -ssd <random%>

Changed the flag -loot to --shuffle-steals-drops <random%>.  If random% = 0, the steals and drops table is just shuffled.  if random% = 100, this recreates the original -loot.

HansGR committed:
menu bugfix

Fix a problem with the menu description.

HansGR commented:
I changed -loot to -ssd <random%> (--shuffle-steals-drops).  This makes it more useful in general and follows behavior of shuffle chests random%.  

I also fixed the tracking menu entry for -cam, which was broken.

HansGR committed:
Fix menu bug 2

asilverthorn commented on args/steal.py:
```diff
@@ -20,19 +27,27 @@ def flags(args):
-    ]
+    result.append(("Chances", steal_chances))
+
+    result.append(("Shuffle steals & drops", args.shuffle_steals_drops))
```
too long:

![image](https://user-images.githubusercontent.com/96998881/230733480-139edb63-81ea-4989-95e1-bff109a41de6.png)

asilverthorn commented on args/steal.py:
```diff
@@ -20,19 +27,27 @@ def flags(args):
+
+    result.append(("Shuffle steals & drops", args.shuffle_steals_drops))
+    if args.shuffle_steals_drops:
+        result.append(("Random Percent:", f"{args.shuffle_steals_drops_random_percent}%"))
```
nit: please remove the ":" to match others

HansGR committed:
--chests-all-miab: respect Phunbaba 3 & clean up boss list

1. -cam now respects -srp3: if -srp3 is called, Phunbaba3 can be found in chests; otherwise, not.
2. Cleaned up the list of bosses in chests.  Removed:
- 63 (Broken Atma)
- 74 (Dummied Umaro)
- 76 (Guardian repeat)
- 88 (Broken Dirt Drgn)
- 111 (White Drgn repeat)

HansGR committed:
--shuffle-steals-drops: exclude drops in final battle

Apparently drops in final battle will cause problems.

asilverthorn commented:
Partially (all but no-random-encounters) merged with 1.3.0 https://github.com/ff6wc/WorldsCollide/pull/64
